test/puppetlabs/trapperkeeper/services/authorization/authorization_service_test.clj

```diff
@@ -102,15 +102,10 @@
      :type "path"}
     :allow-unauthenticated true}
    {:match-request
-    {:path "/puppet-ca/v1/certificate_request"
-     :method :get
-     :type "path"}
-    :allow-unauthenticated true}
-   {:match-request
-    {:path "/puppet-ca/v1/certificate_request"
-     :method :put
-     :type "path"}
-    :allow-unauthenticated true}])
+    {:path   "/puppet-ca/v1/certificate_request"
```

Comment:
I didn't see anything in the tests which exercise this directly, so I wrote the following:

~~~~clojure
(testing "certificate_request"
    (let [app (build-ring-handler default-rules)]
      (let [req (request "/puppet-ca/v1/certificate_request/ca" :head "127.0.0.1" test-domain-cert)
            {:keys [status body]} (app req)]
        (is (= status 403))
        (is (= body "global deny all - no rules matched")))
      (let [req (request "/puppet-ca/v1/certificate_request/ca" :get "127.0.0.1" test-domain-cert)
            {:keys [status]} (app req)]
        (is (= status 200)))
      (let [req (request "/puppet-ca/v1/certificate_request/ca" :put "127.0.0.1" test-domain-cert)
            {:keys [status]} (app req)]
        (is (= status 200)))))
~~~~

Both the ":get" and ":put" tests fail for this.  I believe this is because the {{method}} private function in "authorization_core.clj" assumes that the original method is a string rather than tolerating the ability for it to be a vector.  I changed the implementation of that from:

~~~~clojure
(keyword (get-in config-map [:match-request :method] :any))
~~~~

... to ...

~~~~clojure
(let [method-from-config (get-in config-map [:match-request :method] :any)]
    (if (vector? method-from-config)
      (into [] (map keyword method-from-config))
      (keyword method-from-config)))
~~~~

With this, the tests pass.  There may be a better way to fix this, but, in any event, I think we need to increase the test coverage and ensure that we have a path for doing multiple method rules when rules are created via the `build-rule` path.